neo4j/hierarchy_reader.go

```diff
@@ -53,6 +52,42 @@ func (n *Neo4j) GetHierarchyElement(ctx context.Context, instanceID, dimension,
 	return
 }
 
+// HierarchyExists returns true if the hierarchy exists
+func (n *Neo4j) HierarchyExists(ctx context.Context, instanceID, dimension string) (hierarchyExists bool, err error) {
+	neoStmt := fmt.Sprintf(query.HierarchyExists, instanceID, dimension)
+	logData := log.Data{
+		"fn":             "HierarchyExists",
+		"cypher":         neoStmt,
+		"instance_id":    instanceID,
+		"dimension_name": dimension,
+	}
+
+	var vertices []*models.HierarchyElement
+
+	if vertices, err = n.queryElements(ctx, instanceID, dimension, neoStmt, neoArgMap{}); err != nil {
+		if err == driver.ErrNotFound {
+			hierarchyExists = false
```

Comment:
Is this needed? The default value of hierarchyExists will be false as it is set in the method signature

Reply:
Not technically needed - though I leaned towards being more explicit.

Reply:
Explicit is good, happy for this not to be changed.